.github/workflows/codeql.yml

```diff
@@ -0,0 +1,60 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master", "dev" ]
+  pull_request:
+    branches: [ "master", "dev" ]
+  schedule:
+    - cron: '16 19 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    timeout-minutes: 360
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - SDK: "$NANOS_SDK"
+            artifact: hathor-app-nanoS
+          - SDK: "$NANOX_SDK"
+            artifact: hathor-app-nanoX
+          # - SDK: "$NANOSP_SDK"
+          #   artifact: hathor-app-nanoSP
+        language: [ 'c-cpp' ]
+    container:
+      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-legacy:latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v3
```

Comment:
question: I wanted to suggest a specific commit for `codeql-action@v3`, but couldn't find any v3 [on their releases page](https://github.com/github/codeql-action/releases).

[The latest run](https://github.com/HathorNetwork/hathor-ledger-app/actions/runs/8194327894/job/22410095093) is succeeding, however, so this version _is_ available. I only feel it would be good to tag this action to a specific commit to keep our security best practices.

Reply:
Because this is `codeql-action/init` ([here](https://github.com/github/codeql-action/blob/main/init/action.yml)) not just `codeql-action`.